### PR TITLE
Lint mode docs improvements

### PR DESCRIPTION
**Pull request by bdovaz**
Context https://github.com/oxsecurity/megalinter/issues/2202

@nvuillam I have doubts if it makes sense that this row and the schema enum appears if "cli_lint_mode: file" because in this case there is no other possible value. It would be the equivalent case to when "cli_lint_mode: project" then neither the row nor the schema is generated.

Check everything in case there is a case that I have not understood correctly.

### CODE REVIEW

**Review by nvuillam — COMMENTED** [review]
If cli lint mode == project, only available value in enum should be project, is this taken in account ?

**Comment by nvuillam** [comment]
And json schema must be generated again with bash build.sh :)

**Review by nvuillam — CHANGES_REQUESTED** [review]
Plz check comments :)

**Comment by bdovaz** [comment]
> If cli lint mode == project, only available value in enum should be project, is this taken in account ?

@nvuillam Right now in the original code if it is project that entry does not appear in the table not even: https://github.com/oxsecurity/megalinter/blob/main/.automation/build.py#L1257

It is necessary to decide if in the case of == file as well as in the case of == project if that entry should appear or not with the only value of each case or not. Currently as I say if == project, it does not appear.

**Comment by nvuillam** [comment]
Agreed, if there is only one value possible, no need to make it appear in the variables list in the doc :)

**Comment by bdovaz** [comment]
> And json schema must be generated again with bash build.sh :)

@nvuillam Do you mean "bash build.sh --doc"? Because the only thing I change with this code is documentation nothing else.

If I run "bash build.sh" nothing changes.

**Comment by nvuillam** [comment]
That's strange, add_in_config_schema_file should update the json schema to remove obsolete enum values...

**Comment by bdovaz** [comment]
> That's strange, add_in_config_schema_file should update the json schema to remove obsolete enum values...

@nvuillam But that's what I'm telling you, the code involving what I've already discussed:

https://github.com/oxsecurity/megalinter/blob/main/.automation/build.py#L1257

It is invoked only by adding --doc by generate_documentation() method

**Comment by nvuillam** [comment]
and even when you call bash build.sh --doc it does not update the JSON schema ?

**Comment by bdovaz** [comment]
> and even when you call bash build.sh --doc it does not update the JSON schema ?

@nvuillam Yes, of course in that case yes but originally you had told me to run "bash build.sh" without "--doc" hence the confusion in the conversation I think.

**Comment by nvuillam** [comment]
You acn run with the `--doc` in special cases like this (it just updates more files so create more conflicts, that's why the doc says to run only `bash build.sh` by default :) )

**Comment by bdovaz** [comment]
@nvuillam I have one last doubt.

We have made it clear that neither "cli_lint_mode: file" nor "cli_lint_mode: project" allow to override the *_CLI_LINT_MODE variable and that is why it should not appear in the documentation.

You also said that if "cli_lint_mode: list_of_files" is implicit to allow linting in "cli_lint_mode: file" mode, but what happens in that case with the "project" mode? All linters that have "cli_lint_mode: list_of_files" do not necessarily have to work in "project" mode, right?

**Comment by nvuillam** [comment]
list_of_files is the most optimized way to run linters (because identifying files has already been done by MegaLinter), contrary to file which is performance hell (but can be required for some rare cases like TAP output). For those few cases, user must be able to go from list_of_files to file
default project mode is mostly for linters that can not accept file or list_of_file, like security linters, that's why it's not possible to override the value
Allowing file or list_of_files to be overridden to project is mostly for workarounds, for example with linters that have a problem to find their config file when the current folder is not the repo root (that will be solved in the infinite PR #1877 )

File: megalinter/Linter.py [hunk 1]
```diff
@@ -571,10 +571,10 @@ def load_config_vars(self, params):
                 )
             elif (
                 cli_lint_mode_descriptor == "file"
-                and cli_lint_mode_config == "list_of_files"
+                and (cli_lint_mode_config == "list_of_files" or cli_lint_mode_config == "project")
```
Comment (nvuillam) [inline megalinter/Linter.py:574]: nah, project can work as override of file mode ^^

Comment (bdovaz) [reply megalinter/Linter.py]: Done

**Comment by bdovaz** [comment]
@nvuillam I think it's ready!

**Comment by nvuillam** [comment]
this CI error arrives from time to time, it's ok to bypass it

**Review by nvuillam — APPROVED** [review]

**Pull request merged**